### DISCUSSION TITLE
Does vitest use a transpiler?

### DISCUSSION THREAD

**Discussion by talkohavy** [vitest-dev/vitest#4290] [Q&A]
Does vitest use a transpiler?

I saw some competing testing library that says jest & vitest both use transpilers to process your source code, and that they don't.
Now, I know for a fact that it's true for jest, since they need to transpile es6 to commonjs.... but for vitest?
vitest support module type js out of the box, so why does it need a transpiler? Does it really use one?

**Comment by sheremet-va** [comment]
Yes, it uses Vite to transform files. We at least need to resolve import paths correctly.

**Comment by giltayar** [comment]
But what if the test is testing Node.js code that uses ESM and doesn't need any transpiling? Does it still go through Vite? Can I disable that?

**Reply by sheremet-va** [reply]
It goes through a small transformer to make mocker imports work. You cannot remove "vite" from "vite"st
